processing/e-insert-tables.js

```diff
@@ -0,0 +1,55 @@
+var createReadStream = require('fs').createReadStream;
+var createWriteStream = require('fs').createWriteStream;
+var readdirSync = require('fs').readdirSync;
+var path = require('path');
+var parallel = require('async').parallel;
+
+var baseDir = 'data/processing/d-simplify-props'
+var knex = require('./db/connection/.js')
+var postgis = require('knex-postgis');
+
+// streams to read and write geojsons
+var geojsonStream = require('geojson-stream');
+var parser = geojsonStream.parse();
+var stringifier = geojsonStream.stringify();
+// helps split single-line json into chunked-by-line geojson
+var split = require('split');
+// tmp dir with geojsons
+var adminPath = `${baseDir}/tmp`;
+var admins = readdirSync(adminPath)
+
+var db = knex({dialect: 'postgres'});
+var st = postgis(db);
+
+// create list of async functions to pass to parallel
+const adminTasks = admins.map((admin) => {
+  return function(cb) {
+    var basename = admin.split('-')[1]
+    var adminFile = path.join(adminPath, admin)
+    var adminFile = path.join('./', admin);
+    var adminFileStream = createReadStream(adminFile)
+    .pipe(split())
+    .pipe(parser)
+    .on('data', (feature) => {
+      insertIntoTable(feature, basename)
+    })
+    .on('end', () => { cb(null, null) })
+  }
+});
+
+function insertIntoTable (feature, admin) {
+  const properties = feature.properties;
+  const geometry = feature.geometry;
+  const statement = db.insert({
+    type: admin,
+    id: properties.id,
+    parent_id: properties.p_id,
+    geo: st.geomFromGeoJSON(geometry),
+    name_en: properties.en_name,
+    name_vn: ''
+  }).into(`${admin}-table`).toString();
```

Comment:
Won't all levels of boundaries live in the same admin table? That's how I have it set up, and how we did it for ORMA Philippines. With one table, it's easier to query, and self-joins make it as versatile as multiple tables (eg, `SELECT child.name_en FROM admin_areas AS parent JOIN admin_areas AS child ON parent.id=child.parent_id WHERE parent.id='12345'`). Plus, then it's portable to different countries (which would have different names for their admin levels).

Reply:
For sure they can all live in the same table / they need to if that is how you have things set up. I'll make the changes to that script we're doing the right insert.